### PR TITLE
[FLINK-28967] Fix Hive multi-version support for Table Store

### PR DESCRIPTION
**Pull request by tsreaper**
Currently table store Hive catalog and connector can only work for Hive 2.3. Support for Hive 2.1 and 2.2 should be fixed.

### CODE REVIEW

File: docs/content/docs/engines/hive.md [hunk 1]
```diff
@@ -32,20 +32,51 @@ Table Store currently supports the following features related with Hive:
 
 ## Version
 
-Table Store currently supports Hive 2.3.
+Table Store currently supports Hive 2.1, 2.2 and 2.3.
```
Comment (JingsongLi) [inline docs/content/docs/engines/hive.md:35]: Add hive 2.1 2.2 in engines/overview too.

File: docs/content/docs/engines/hive.md [hunk 2]
```diff
@@ -32,20 +32,51 @@ Table Store currently supports the following features related with Hive:
...
 ## Execution Engine
 
 Table Store currently supports MR and Tez execution engine for Hive.
 
 ## Install
 
+### Table Store Hive Catalog (For Flink)
+
+{{< stable >}}
+Download the jar file with corresponding version.
```
Comment (JingsongLi) [inline docs/content/docs/engines/hive.md:46]: Add some information to spark3 page too?
Spark3 need this to create table and etc...

File: tools/releasing/create_binary_release.sh [hunk 3]
```diff
@@ -56,23 +56,39 @@ mkdir ${RELEASE_DIR}
...
 
 cd ${RELEASE_DIR}
-gpg --armor --detach-sig "flink-table-store-dist-$RELEASE_VERSION.jar"
-gpg --armor --detach-sig "flink-table-store-hive-connector-$RELEASE_VERSION.jar"
-gpg --armor --detach-sig "flink-table-store-spark-$RELEASE_VERSION.jar"
-gpg --armor --detach-sig "flink-table-store-spark2-$RELEASE_VERSION.jar"
+gpg --armor --detach-sig "flink-table-store-dist-${RELEASE_VERSION}.jar"
+gpg --armor --detach-sig "flink-table-store-hive-connector-${RELEASE_VERSION}.jar"
+gpg --armor --detach-sig "flink-table-store-spark-${RELEASE_VERSION}.jar"
+gpg --armor --detach-sig "flink-table-store-spark2-${RELEASE_VERSION}.jar"
```
Comment (JingsongLi) [inline tools/releasing/create_binary_release.sh:87]: You need to add gpg and sha for Hive 2.1 and 2.2.

**Comment by JingsongLi** [comment]
Do we really need `flink-table-store-hive-catalog_1.14.jar`? It should not have the flink code in it?

**Comment by SteNicholas** [comment]
@tsreaper @JingsongLi , does the NOTICE file of this module need to be updated?

**Comment by SteNicholas** [comment]
@tsreaper, the Hive 2.1 and 2.2 use the latest version 2.1.1 and 2.2.0, hence does the Hive 2.3 need to support the latest version 2.3.9?

**Review by JingsongLi — APPROVED** [review]
Looks good to me!

**Pull request merged**